packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_performance.py

```diff
@@ -0,0 +1,79 @@
+import time
+import numpy as np
+import plotly.graph_objs as go
+
+np.random.seed(1)
+
+
+def test_performance_b64_scatter3d():
+    N = 10000
+
+    x = np.random.randn(N)
+    y = np.random.randn(N).astype("float32")
+    z = np.random.randint(size=N, low=0, high=256, dtype="uint8")
+    c = np.random.randint(size=N, low=-10, high=10, dtype="int8")
+
+    # Test the performance with lists
+    list_start = time.time()
+    fig = go.Figure(
+        data=[
+            go.Scatter3d(
+                x=x.tolist(),
+                y=y.tolist(),
+                z=z.tolist(),
+                marker=dict(color=c.tolist()),
+                mode="markers",
+                opacity=0.2,
+            )
+        ]
+    )
+    list_time_elapsed = time.time() - list_start
+
+    # Test the performance with base64 arrays
+    np_start = time.time()
+    fig = go.Scatter3d(
+        x=x,
+        y=y,
+        z=z,
+        marker=dict(color=c),
+        mode="markers",
+        opacity=0.2,
+    )
+    np_time_elapsed = time.time() - np_start
+
+    # np should be faster than raw
+    assert (np_time_elapsed / list_time_elapsed) < 0.75
```

Comment:
Could we further reduce the 0.75 ratio here?

Reply:
I did some tests and it's hovering around .7 so I think that's as low as it'll go